torch/csrc/jit/script/lexer.h

```diff
@@ -185,6 +186,36 @@ struct SharedParserData {
     *len = endptr - startptr;
     return *len > 0;
   }
+
+  bool isCharCount(char c, const std::string& str, size_t start, int len) {
+    //count checks from [start, start + len)
+    return start + len <= str.size() && std::count(str.begin() + start, str.begin() + start + len, c) == len;
+  }
+
+  // python conconcatenates all adjacent strings "a" "b" == "ab"
+  // strings can be enclosed with 1 or 3 single or double quotes
+  // if enclosed 3 quotes newlines are valid
+  // as elsewhere, backslash and new line should be ignored
+  bool isString(const std::string& str, size_t start, size_t* len) {
+    char quote = str[start];
+    if (quote != '\"' && quote != '\'')
+      return false;
+    int quote_len = isCharCount(quote, str, start, 3) ? 3 : 1;
+    size_t end = start + quote_len;
+    while(end < str.size() && !isCharCount(quote, str, end, quote_len)) {
+      if (str[end] == '\n' && quote_len != 3) {
+        return false;
+      }
+      //handle escaped characters
+      if (str[end] == '\\') {
+        end++;
+      }
+      end++;
+    }
+    *len = end - start + quote_len;
+    return end < str.size();
```

Comment:
I'm not 100% sure about this condition, because it looks like this trivial program would get rejected:
```
x = "adsf"
```

Reply:
I've tested out that string and it works. But you're right I will right invariants out.

Reply:
that program will get accepted because end is at position 5 and the str is length 6. I will comment more thoroughly.